lib/asset/AssetGenerator.js

```diff
@@ -113,6 +113,8 @@ class AssetGenerator extends Generator {
 				} else {
 					const assetModuleFilename =
 						this.filename || runtimeTemplate.outputOptions.assetModuleFilename;
+					let assetModulePublicPath =
+						runtimeTemplate.outputOptions.assetModulePublicPath;
```

Comment:
```suggestion
						this.publicPath || runtimeTemplate.outputOptions.assetModulePublicPath;
```

I also want to have the `publicPath` in the generator options similar to `filename`.

`module.generator.asset.publicPath` or `Rule.generator.publicPath`. The former one also need to be added to the schema.